src/main/java/com/mongodb/jdbc/MongoSQLStatement.java

```diff
@@ -31,10 +34,24 @@ public ResultSet executeQuery(String sql) throws SQLException {
                 iterable = iterable.batchSize(fetchSize);
             }
 
-            resultSet = new MongoSQLResultSet(this, iterable.cursor());
+            MongoJsonSchemaResult schemaResult =
+                    currentDB
+                            .withCodecRegistry(MongoDriver.registry)
+                            .runCommand(getSchemaCmd, MongoJsonSchemaResult.class);
+
+            MongoJsonSchema schema = schemaResult.schema.jsonSchema;
```

Comment:
If the command fails, the java driver throws an exception. If it succeeds, the `schemaResult` object will definitely be non-null. The `schema` field should also never be null ([docs](https://docs.mongodb.com/datalake/reference/cli/sql/sqlgetschema/#output)). 

The `jsonSchema` field _may_ be null if the collection has no schema. I'm not sure what the java driver does when unmarshaling an object with missing fields.

If the collection has no schema, the response is
```
{ "ok" : 1, "metadata" : { }, "schema" : { } }
```
So the `schema` field would probably be initialized with default values: `version = 0`, `jsonSchema = null`? Worth investigating how the java driver handles this kind of case (we can test this without using the `sqlGetSchema` command). If the `jsonSchema` could be null, we should definitely handle that.

Reply:
I'll test this, since I already have a framework for it

Reply:
It maintains the `jsonSchema` field but with a null value.  Which, in retrospect, should be obvious, otherwise it would fail to deserialize to a `JsonSchemaResult` haha.